### PR TITLE
SDI-255: ✨ Use restore info to determine if refresh is required

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: helm_deploy/prisoner-offender-search/restore.sh [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
 
-# Snapshot restores only happen every other week, crontab doesn't really support this
-week=$(date +%U)
-if [[ $(("$week" % 2)) == 0 && -z "$FORCE_RUN" ]]; then
-  echo "Not running restore this week, restores only happens on odd week numbers, check back next week."
-  exit 0
+check_http() { http --stream --check-status --ignore-stdin --timeout=600 "$@"; }
+
+# grab last restore date from Prison API
+if ! DATABASE_RESTORE_INFO=$(check_http GET "$PRISON_API_BASE_URL/api/restore-info"); then
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/restore.sh:7]: Prison API will return either a `404` or a json date

File: helm_deploy/prisoner-offender-search/restore.sh [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
+  echo -e "\nUnable to find any restore information.  For environments other than pre-prod this is expected"
+  if [[ -z "$FORCE_RUN" ]]; then
+    echo -e "\nTo force a run set the FORCE_RUN environment variable when creating the job (see README.md)"
+    echo "$DATABASE_RESTORE_INFO"
+    exit 0
+  fi
 fi
+DATABASE_RESTORE_DATE=$(echo "$DATABASE_RESTORE_INFO" | jq -r .)
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/restore.sh:15]: need to strip the quotes off the date so can compare

File: helm_deploy/prisoner-offender-search/templates/elasticsearch-restore.yaml [hunk 3]
```diff
@@ -17,7 +17,7 @@ metadata:
   labels:
     {{- include "app.labels" . | nindent 4 }}
 spec:
-  schedule: "30 3 * * 0"
+  schedule: "30 */4 * * *"
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/templates/elasticsearch-restore.yaml:20]: schedule to run every 4 hours every day

File: helm_deploy/prisoner-offender-search/templates/elasticsearch-restore.yaml [hunk 4]
```diff
@@ -62,6 +62,8 @@ spec:
                   fieldPath: metadata.namespace
             - name: NAMESPACE_OVERRIDE
               value: {{ .Values.cron.snapshotNamespace }}
+            - name: PRISON_API_BASE_URL
+              value: {{ index .Values "generic-service" "env" "API_BASE_URL_NOMIS" }}
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/templates/elasticsearch-restore.yaml:66]: need prison api url to grab the latest restore date

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**